src/pip/_internal/operations/build/wheel.py

```diff
@@ -23,13 +24,33 @@ def build_wheel_pep517(
     try:
         logger.debug("Destination directory: %s", tempd)
 
-        runner = runner_with_spinner_message(f"Building wheel for {name} (PEP 517)")
+        artifact = "editable" if editable else "wheel"
+        pep = "660" if editable else "517"
+        runner = runner_with_spinner_message(
+            f"Building {artifact} for {name} (PEP {pep})"
+        )
         with backend.subprocess_runner(runner):
-            wheel_name = backend.build_wheel(
-                tempd,
-                metadata_directory=metadata_directory,
-            )
+            if editable:
+                try:
+                    wheel_name = backend.build_editable(
+                        tempd,
+                        metadata_directory=metadata_directory,
+                    )
+                except HookMissing as e:
+                    logger.warning(
+                        "Cannot build %s %s because the build "
+                        "backend does not have the %s hook",
+                        artifact,
+                        name,
+                        e,
+                    )
```

Comment:
I need to check if this failure could be trigger a fallback to setup.py develop. Normally not because the new resolver guarantees that generate_metadata has been run before building ?

Reply:
Yep, that's really an error because generate_metadata will have set supports_pep660 first.